frontend/src/AlertsManagement/History.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import Box from '@material-ui/core/Box';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import Typography from '@material-ui/core/Typography';
+import { getAlertsData } from './management_helpers';
+import { formatDate } from '../time_utils';
+
+class History extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      allAlerts: null,
+    }
+  }
+
+  async componentDidMount() {
+    var results = await getAlertsData();
+    if (results.length !== 3) {
+      throw new Error("getAlertsData() did not return the correct alerts data.")
```

Comment:
This error would happen if there was something wrong that happened in getAlertsData, so I think it should be after calling getAlertsData. Does that make sense?

Reply:
I see since the `getAlertsData` calls fetch, so you have error inside `getAlertsData` for the fetch, but this results.length !== 3 is checking the creation of the `results` that is being done in the `getAlertsData`.

ie for example there's changes made to the `getAlertsData` API and so results length is not 3 anymore. From my recall, you're calling `getAlertsData` in some other place as well right not just in the `History`. Some ways to go around this: have a validateAlertsDataResult that can be called for one consistent place for data validation, this way the History can just call that